datalad/distributed/ora_remote.py

```diff
@@ -498,18 +560,21 @@ def get(self, src, dst, progress_cb):
                     progress_cb(bytes_received)
 
     def rename(self, src, dst):
-        self._run('mv {} {}'.format(sh_quote(str(src)), sh_quote(str(dst))))
+        with self.ensure_writeable(dst.parent):
+            self._run('mv {} {}'.format(sh_quote(str(src)), sh_quote(str(dst))))
 
     def remove(self, path):
         try:
-            self._run('rm {}'.format(sh_quote(str(path))))
+            with self.ensure_writeable(path.parent):
+                self._run('rm {}'.format(sh_quote(str(path))), check=True)
         except RemoteCommandFailedError as e:
-            raise RIARemoteError(
-                "Write permissions for a key's parent"
-                "directory are also required to drop content.") from e
+            raise RIARemoteError(f"Unable to remove {path}. Could not "
+                                 "obtain write permission for containing"
+                                 "directory.") from e
```

Comment:
```suggestion
            raise RIARemoteError(f"Unable to remove {path} "
                                 "or to obtain write permission in parent directory.") from e
```

Thinking about this more, I'd say this is the better message. Without further inspection, we do not know whether this failure is about write permissions. Unlike before, the path reported in the exception will be the problematic one directly, and not just some proxy that requires expert knowledge to make the connection to interpret.